airflow/models/taskinstance.py

```diff
@@ -3000,12 +3019,12 @@ def _execute_task(self, context: Context, task_orig: Operator):
         return _execute_task(self, context, task_orig)
 
     @provide_session
-    def defer_task(self, exception: TaskDeferred, session: Session) -> None:
-        """Mark the task as deferred and sets up the trigger that is needed to resume it.
+    def defer_task(self, exception: TaskDeferred | None = None, session: Session = NEW_SESSION) -> None:
+        """Mark the task as deferred and sets up the trigger that is needed to resume it when TaskDeferred is raised.
 
         :meta: private
         """
-        _defer_task(ti=self, exception=exception, session=session)
+        _defer_task(ti=self, session=session, exception=exception)
```

Comment:
```suggestion
        _defer_task(ti=self, exception=exception, session=session)
```

nit: this order matches the signature.

Reply:
Updated. Thanks!